R/model.R

```diff
@@ -121,14 +121,37 @@ fall_run_model <- function(scenario = NULL, mode = c("seed", "simulate", "calibr
       ..surv_egg_to_fry_int = ..params$..surv_egg_to_fry_int
     )
 
-    min_spawn_habitat <- apply(..params$spawning_habitat[ , 10:12, year], 1, min)
+    min_spawn_habitat <- apply(..params$spawning_habitat[ , 3:6, year], 1, min)
 
-    accumulated_degree_days <- cbind(oct = rowSums(..params$degree_days[ , 10:12, year]),
-                                     nov = rowSums(..params$degree_days[ , 11:12, year]),
-                                     dec = ..params$degree_days[ , 12, year])
+    accumulated_degree_days <- cbind(march = rowSums(..params$degree_days[ , 3:6, year]),
+                                     april = rowSums(..params$degree_days[ , 4:6, year]),
+                                     may = rowSums(..params$degree_days[ , 5:6, year]),
+                                     june = ..params$degree_days[ , 6, year])
 
     average_degree_days <- apply(accumulated_degree_days, 1, weighted.mean, ..params$month_return_proportions)
 
+    prespawn_survival <- surv_adult_prespawn(average_degree_days,
+                                             ..surv_adult_prespawn_int = ..params$..surv_adult_prespawn_int,
+                                             .deg_day = ..params$.adult_prespawn_deg_day)
+
+    # Apply SR pools logic
+    init_adults <- rbinom(31, round(init_adults), prespawn_survival)
+
+    # spring run above capacity die, capacity based on spring run pools
+    init_adults <- ifelse(init_adults >= ..params$spring_run_pools, ..params$spring_run_pools, init_adults)
```

Comment:
Currently a warning here. `init_adults` has a length of 31 (all watersheds) `spring_run_pools` currently does not include the sutter or yolo bypass. I created a new PR in `DSMhabitat`, [`pools`](https://github.com/CVPIA-OSC/DSMhabitat/pull/23/files) that addresses this problem by adding in Yolo and Sutter bypasses to pools.